src/components/DateTimePicker/index.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useMemo, useRef, useState } from 'react';
+import ReactDatePicker, { DatePickerProps } from 'react-datepicker';
+import { useTranslation } from 'react-i18next';
+import { Alert, Button, Icon, Tooltip } from '@trussworks/react-uswds';
+import classNames from 'classnames';
+
+import { formatDateUtc, isDateInPast } from 'utils/date';
+
+import 'react-datepicker/dist/react-datepicker.css';
+import './index.scss';
+
+type DateTimePickerProps = DatePickerProps & {
+  id: string;
+  name: string;
+  alertIcon?: boolean; // Whether to show the warning icon
+  alertText?: boolean; // Whether to show the warning text. Sometimes we want to render the warning text under a different parent/UI element - outside the scope of this component
+  className?: string;
+};
+
+/*
+  This component is a wrapper around the ReactDatePicker component.
+  It is used to display a date picker with an optional warning icon and alert. Default is true
+  The warning icon and alert are only displayed if the date is in the past.
+*/
+const DateTimePicker = ({
+  id,
+  name,
+  alertIcon = true,
+  alertText = true,
+  className,
+  ...props
+}: DateTimePickerProps) => {
+  const { t: generalT } = useTranslation('general');
+
+  const [isOpen, setIsOpen] = useState(false);
+
+  const datePickerRef = useRef<ReactDatePicker>(null);
+
+  // Check if the date is in the past
+  const dateIsInPast = useMemo<boolean>(() => {
+    if (props.value) {
+      return isDateInPast(props.value);
+    }
+    return false;
+  }, [props.value]);
+
+  return (
+    <div>
+      <div className={classNames('display-flex margin-top-1', className)}>
+        <ReactDatePicker
+          {...props}
+          ref={datePickerRef}
+          id={id}
+          name={name}
+          open={isOpen}
+          onClickOutside={() => setIsOpen(false)}
+          onSelect={() => setIsOpen(false)}
+          selected={props.value ? new Date(props.value) : null}
+          value={
+            typeof props.value === 'string'
+              ? formatDateUtc(props.value, 'MM/dd/yyyy')
+              : props.value
+          }
```

Comment:
I think using the `isClearable` prop helps, but it still seems weird to me to not be able to use backspace at all in a text input.

I looked at the docs and played around with the implementation a bit, and it looks like not explicitly setting the `value` prop here on the `ReactDatePicker` component fixes the issue. The `selected` prop still sets the selected date when the field has a default value.

I haven't tested it thoroughly enough to know if that causes other issues, though.